test/conv_common.hpp

```diff
@@ -1646,6 +1660,36 @@ struct conv_driver : test_driver
 
     void run()
     {
+        if(input.desc.GetSize() != in_layout.size() ||
+           weights.desc.GetSize() != fil_layout.size() || input.desc.GetSize() != out_layout.size())
+        {
+            MIOPEN_LOG_E("layout not match dimension size!");
+        }
+
+        // replace desc to set layout information here
+        if(in_layout != "NCHW" || in_layout != "NCDHW")
```

Comment:
Please explain why this is required.

Reply:
the code under `if` is to re-construct `input.desc`, `input.desc` is first constructed by its constructor [here](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/test/conv2d.cpp#L33), and the `tensor` constructor seems very hard to support different layout only from command line.

Reply:
Now I see. Could you please add this reasoning in C++ comments.

Reply:
OK, done in https://github.com/ROCmSoftwarePlatform/MIOpen/pull/765/commits/ee0302f0bd4ffe149ab55fbb80437773ab7cf148